app/form_persisters/form_persister.rb

```diff
@@ -25,37 +20,63 @@ def save_all(forms:)
     end
   end
 
-  private
+  def instance(form)
+    Instance.new(form: form, form_persister: self)
+  end
 
-    def before_save(form:)
-      create_files(form: form)
+  class Instance
+    attr_reader :form, :form_persister
+    delegate :adapter, :storage_adapter, to: :form_persister
+    delegate :persister, :query_service, to: :adapter
```

Comment:
I'm not sure this change is going to go in, given @jcoyne's comment above. That being said, for discussion:

I get where you're coming from, the issue is that the API of this and of `persister` are basically the same - just the named argument is different. And it's done that way on purpose, to help the developer.